ministryofjustice/laa-apply-for-criminal-legal-aid
# [CRIMAPP-912] Display benefit check outcome for unemployed client flow

## Description of change
Displays benefit check outcome for the unemployed client flow, updating the review, completed and returned application screens
Moves the benefit type attributes from the client details section to the new passporting benefit check section 
Amends design of the card to display a single change link for the benefit section
Submits and rehydrates the benefit check outcome attributes 

Edit: Introduces an attribute `benefit_check_status` which is calculated upon submission and represents the dwp outcome per dwp check journey and this is submitted to review. This will make displaying the outcome easier in review but also differentiates between `benefit_check_result` which is simply the result of the dwp check and can only take three values `true` `false` `nil` to the labels that are displayed on apply which is not just the result of the check but also takes into account the responses to the other questions in the dwp check flow e.g. `has_benefit_evidence`

## Link to relevant ticket
[CRIMAPP-912](https://dsdmoj.atlassian.net/browse/CRIMAPP-912)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

### After changes:
Screenshots of the new passporting benefit section for different scenarios 
<img width="1081" alt="Screenshot 2024-05-22 at 12 26 35" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/1bcc03eb-e3c1-456a-a94a-210a20451f99">
<img width="1081" alt="Screenshot 2024-05-22 at 12 24 57" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/b813292d-9514-4672-97e2-f28ff8f90a53">
<img width="1081" alt="Screenshot 2024-05-22 at 12 22 45" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/a467bf61-8b3e-4077-97c4-779300ce1b9b">
<img width="1081" alt="Screenshot 2024-05-22 at 12 19 17" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/bfb18806-48ec-482e-9c95-e56e679e7b53">
<img width="1081" alt="Screenshot 2024-05-22 at 12 54 56" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/48f5a78c-ce7d-432a-b83b-89d086691d17">
<img width="1081" alt="Screenshot 2024-05-22 at 12 34 23" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/6c512094-c92d-4be6-aab7-20edf6fcdd3e">
<img width="1081" alt="Screenshot 2024-05-22 at 12 28 43" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/d0c310bd-c586-4ac6-abfc-404466ebcb02">


## How to manually test the feature
Requires [datastore pr](https://github.com/ministryofjustice/laa-criminal-applications-datastore/pull/264) to verify submission.

Test the scenarios outlined in the [ticket](https://dsdmoj.atlassian.net/browse/CRIMAPP-912)/[figma](https://www.figma.com/design/thkvnDBbQbHlSqzt1TE3lk/Crime-Apply?node-id=10335-85981&t=XHpTZgeMrrckBs4q-0)

You can also confirm the view for the following application types
- PSE application -> does not display 
- Appeal no changes application -> does not display
- Under 18 application -> does not display
- Historical application -> does display section but does not display the benefit check rows 
- Returned application that was passported on means -> does display the section and the confirmed dwp result

[CRIMAPP-912]: https://dsdmoj.atlassian.net/browse/CRIMAPP-912?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
